packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -224,12 +224,6 @@ library LibProving {
 
         if (tier.contestBond == 0) {
             assert(tier.validityBond == 0);
-            // When contestBond is zero for the current tier, it signifies
-            // it's the top tier. In this case, it can overwrite existing
-            // transitions without contestation.
```

Comment:
because of this code:
```ts
 ts.blockHash = tran.blockHash;
            ts.signalRoot = tran.signalRoot;
            ts.prover = msg.sender;

            if (ts.contester != address(0)) {
                // At this point we know that the contester was right
                tko.transfer(ts.contester, ts.validityBond >> 2 + ts.contestBond);
                ts.contester = address(0);
                ts.validityBond = 0;
            }
            ```
            
            even though the contester was wrong, he will be rewarded, and the guardian prover will now be the prover of the block, and get the original provers reward? so anyone can contest any block, to steal proof money.

Reply:
i think we need this:
```solidity
if (ts.contester != address(0) && tran.blockHash != ts.blockHash && tran.signalRoot != ts.signalRoot) {
                // At this point we know that the contester was right
                tko.transfer(ts.contester, ts.validityBond >> 2 + ts.contestBond);
                ts.contester = address(0);
                ts.validityBond = 0;
            } else {
               // we know the contester was wrong, penalizse them or something
            }
 ```

Reply:
also this block further down:
```ts
emit TransitionProved({
                blockId: blk.blockId,
                tran: tran,
                prover: msg.sender,
                validityBond: 0,
                tier: proof.tier
            });
            ```
            
i think we dont want to emit this event, in this case, we should have a new event like `ContestationDismissed` or something, in the case of the contester being wrong and the guardian prover proving the original proof is correct.

Reply:
And perhaps we should charge a user to stake something to contest, so they can be penalized when they contest incorrectly, and trying to attack/halt chain.

Reply:
> i think we need this:
> 
> ```solidity
> if (ts.contester != address(0) && tran.blockHash != ts.blockHash && tran.signalRoot != ts.signalRoot) {

> ```

Yes, almost.. :) We need to move that check one side inner, but good observation i changed it !

Reply:
> i think we dont want to emit this event, in this case, we should have a new event like `ContestationDismissed` or something, in the case of the contester being wrong and the guardian prover proving the original proof is correct.

Well, as everything is retrievable with the transaction(s) - like it is seen it is a tier guardian, i think it is fine as is, but can add that if it helps!